src/keyedByPropType.js

```diff
@@ -0,0 +1,42 @@
+export default function keyedByPropType(validatorFn, name = 'keyedByPropType') {
+  const isObject = function isObject(o) {
+    return typeof o === 'object' && o !== null && !Array.isArray(o);
```

Comment:
i wonder if this would be better as just `keysOf`, and have it not check that it's an object? `objectOf` and `PropTypes.object` can be composed to ensure this if desired, and you might want, for example, a function with specific extra properties (like "propTypes" if it's an SFC).